app/backend/lib/awsCommon.ts

```diff
@@ -1,31 +1,39 @@
-import https from 'https'; 
+import http from 'https';
+import { NodeHttpHandler } from '@aws-sdk/node-http-handler';
 import {
   fromTemporaryCredentials,
   fromEnv,
 } from '@aws-sdk/credential-providers';
-
 import config from '../../config';
+import CustomHttpsAgent from './CustomHttpsAgent';
 
-const AWS_S3_REGION = config.get('AWS_S3_REGION'); 
-const AWS_ROLE_ARN = config.get('AWS_ROLE_ARN'); 
+const AWS_S3_REGION = config.get('AWS_S3_REGION');
+const AWS_ROLE_ARN = config.get('AWS_ROLE_ARN');
 
-const agent = new https.Agent({
+const httpsAgent = new CustomHttpsAgent();
+const httpAgent = new http.Agent({
   maxSockets: 300,
-  keepAlive: true
- });
+  keepAlive: true,
+});
+
+const nodeHandler = new NodeHttpHandler({
+  httpAgent,
+  httpsAgent,
+});
 
 const awsConfig = {
   region: AWS_S3_REGION,
   logger: console,
   httpOptions: {
```

Comment:
I think the httpOptions are deprecated on v3 of the client, if anything adding the type of `S3ClientConfig` to the object can help with that. `retryDelayOptions` is also not available along with maxRetries (I think was renamed to maxAttempts)